guppylang/checker/cfg_checker.py

```diff
@@ -222,7 +219,11 @@ def check_bb(
     if bb == cfg.entry_bb:
         assert len(bb.predecessors) == 0
         for x, use in bb.vars.used.items():
-            if x not in cfg.ass_before[bb] and x not in globals:
+            if (
+                x not in cfg.ass_before[bb]
+                and x not in globals
+                and x not in generic_params
+            ):
```

Comment:
There are a bunch of places where we forgot to check if variables are generic parameters. This used to be ok since generic params were also globals (since you declare them globally via `guppy.type_var`), but now we also have locally scoped ones